PyPoE/poe/file/bundle.py

```diff
@@ -509,6 +528,27 @@ def _read(self, buffer: BytesIO):
                     directory_record.offset + directory_record.size
                 ]
             )
+        
+        self.hash_algorithm = None
+        if len(dirs := self.directories) > 0:
+            dir_iter = iter(dirs.values())
+            root_entry = next(dir_iter)
+            if root_entry.hash == 0x07e47507b4a92e53:
+                self.hash_algorithm = HASH_ALGORITHM.FNV1A64
+            else:
+                h = root_entry.hash
+                # Find seed from root directory hash via inverses
+                h ^= h >> 47
+                h = (h * 0x5F7A0EA7E59B19BD) % 2**64
+                h ^= h >> 47
+                for cand in dir_iter:
+                    if len(cand.paths) > 0:
+                        dir = cand.paths[0].rsplit('/', 1)[0]
+                        if _hash_path_3_21_2(dir, h) == cand.hash:
+                            self.hash_algorithm = HASH_ALGORITHM.MURMURHASH64A
+                            self.hash_seed = h
+                        break
```

Comment:
is this loop really necessary? in the end it sets the hash_algorithm and seed to whatever was calculated before it. 
I assume the idea is to check whether the seed is actually valid, but if thats the case and its not valid it probably should raise some exception instead of failing later on since we dont check for None in get_hash or at any other point

Reply:
It is indeed a sanity check for the case where the algorithm or structure may change in the future. I have indeed not thought about the failure mode, bailing out directly may be more elegant.